app/controllers/api/v1/lobbies_controller.rb

```diff
@@ -35,13 +35,7 @@ def answer
 
         return head :not_found if player.blank?
 
-        lobby = Lobby.find(params[:id])
-        question = lobby.quiz.questions.find_by(order: lobby.current_question_index)
-
-        create_player_answers!(player, question)
-        notify_answer_count(lobby)
-
-        @answers = question.answers
+        @answers = AnswerQuestion.new.call(lobby_id: params[:id], player: player, answers: params[:answers]).answers
```

Comment:
Don't send the whole `player` here. Only send the player's `id`!

Make services as easy to use as possible for consumers. If you can find information directly in the service, do that, instead of requiring the consumer to come up with it.

Reply:
But if we've already queried the database for a player, isn't it better for performance to send the player to service and skip a new query inside the service?

Reply:
That's a fair point. Sometimes trading some application performance for development performance is acceptable. In our case, this shouldn't be a big issue.

Rails also helps us in this regard - database hits are cached per request - and you can see this in the terminal if you look at the SQL logs 😄